registry.go

```diff
@@ -0,0 +1,220 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// RegistryService handles communication with the container registry related methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryService struct {
+	client *Client
+}
+
+// RegistryRepository represents a GitLab content registry repository.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryRepository struct {
+	ID        int        `json:"id"`
+	Name      string     `json:"name"`
+	Path      string     `json:"path"`
+	Location  string     `json:"location"`
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+}
+
+func (s RegistryRepository) String() string {
+	return Stringify(s)
+}
+
+// RegistryRepositoryTag represents a GitLab registry image tag.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryRepositoryTag struct {
+	Name     string `json:"name"`
+	Path     string `json:"path"`
+	Location string `json:"location"`
+}
+
+func (s RegistryRepositoryTag) String() string {
+	return Stringify(s)
+}
+
+// RegistryRepositoryTagDetail represents a GitLab registry tag detail.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/container_registry.html
+type RegistryRepositoryTagDetail struct {
+	Name          string     `json:"name"`
+	Path          string     `json:"path"`
+	Location      string     `json:"location"`
+	Revision      string     `json:"revision"`
+	ShortRevision string     `json:"short_revision"`
+	Digest        string     `json:"digest"`
+	CreatedAt     *time.Time `json:"created_at,omitempty"`
+	TotalSize     int        `json:"total_size"`
+}
```

Comment:
I wonder if we should use two different structs for `RegistryRepositoryTag` and `RegistryRepositoryTagDetail`? I would suggest to just use on struct instead. What do you think?

Reply:
I think it makes sense. `RegistryRepositoryTag` is a subset of `
RegistryRepositoryTagDetail`. 👍